parsons/notifications/sendmail.py

```diff
@@ -0,0 +1,259 @@
+# Adapted from Gmail API tutorial https://developers.google.com/gmail/api
+from email.mime.text import MIMEText
+from email.mime.multipart import MIMEMultipart
+from email.mime.image import MIMEImage
+from email.mime.audio import MIMEAudio
+from email.mime.base import MIMEBase
+from email.mime.application import MIMEApplication
+from email.encoders import encode_base64
+from email.utils import parseaddr
+from validate_email import validate_email
+import io
+import logging
+import mimetypes
+import os
+
+# BUG: can't send files equal to or larger than 6MB
+# There is a possible fix
+# `BUG-1` are the changes that would need to be made
+
+# BUG-1
+# import httplib2shim
+# import sys
+
+logger = logging.getLogger(__name__)
+
+
+class SendMail(object):
+    """SendMail base class for sending emails.
+    """
+
+    log = logger
+
+    def send_message(self, message):
+        raise NotImplementedError("send_message is how to send the prepared message")
+
+    def _prepare_message(self, message, message_type):
+        """Convert a email.mime-type object into an object format suitable for your backend to send
+           Consider overriding this, if necessary for your mail-sending backend
+        """
+        return message
+
+    def create_message_simple(self, sender, to, subject, message_text):
```

Comment:
I could go either way, but I think I'd prefer these `create_message_` to be prefixed with the `_`.. just because I can't imagine anyone actually calling them outside of the class. Thoughts? I know that these were pre-existing methods, but it feels like a good time to clean up the API.